frontend/src/components/panels/Document.vue

```diff
@@ -312,6 +328,23 @@ export default defineComponent({
 			if (rulerHorizontal) rulerHorizontal.handleResize();
 			if (rulerVertical) rulerVertical.handleResize();
 		},
+		async pasteFile(e: DragEvent) {
+			const { dataTransfer } = e;
+			if (!dataTransfer) return;
+			e.preventDefault();
+
+			for (let index = 0; index < dataTransfer.items.length; index += 1) {
+				const item = dataTransfer.items[index];
```

Comment:
```js
dataTransfer.items.forEach((item) => {
```
would be preferred

Reply:
A DataTransferItemList does not have a `forEach` method for some reason: https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItemList

Reply:
Can you wrap it in `Array.from(...)`?